pkg/auth/config/secret_data.go

```diff
@@ -0,0 +1,195 @@
+import (
+	"github.com/skygeario/skygear-server/pkg/validation"
+)
+
+var _ = SecretConfigSchema.Add("DatabaseCredentials", `
+{
+	"type": "object",
+	"properties": {
+		"database_url": { "type": "string" },
+		"database_schema": { "type": "string" }
+	},
+	"required": ["database_url"]
+}
+`)
+
+type DatabaseCredentials struct {
+	DatabaseURL    string `json:"database_url,omitempty"`
+	DatabaseSchema string `json:"database_schema,omitempty"`
+}
+
+func (c *DatabaseCredentials) SetDefaults() {
+	if c.DatabaseSchema == "" {
+		c.DatabaseSchema = "public"
+	}
+}
+
+var _ = SecretConfigSchema.Add("RedisCredentials", `
+{
+	"type": "object",
+	"properties": {
+		"host": { "type": "string" },
+		"port": { "type": "integer" },
+		"password": { "type": "string" },
+		"db": { "type": "integer" },
+		"sentinel": { "$ref": "#/$defs/RedisSentinelConfig" }
+	}
+}
+`)
+
+type RedisCredentials struct {
+	Host     string               `json:"host,omitempty"`
+	Port     int                  `json:"port,omitempty"`
+	Password string               `json:"password,omitempty"`
+	DB       int                  `json:"db,omitempty"`
+	Sentinel *RedisSentinelConfig `json:"sentinel,omitempty"`
+}
+
+func (c *RedisCredentials) SetDefaults() {
+	if c.Port == 0 {
+		c.Port = 6379
+	}
+}
+
+func (c *RedisCredentials) Validate(ctx *validation.Context) {
+	if c.Sentinel.Enabled {
+		if len(c.Sentinel.Addrs) == 0 {
+			ctx.Child("sentinel", "addrs").EmitErrorMessage("redis sentinel addrs are not provided")
+		}
+	} else {
+		if c.Host == "" {
+			ctx.Child("host").EmitErrorMessage("redis host is not provided")
+		}
+	}
+}
+
+var _ = SecretConfigSchema.Add("RedisSentinelConfig", `
+{
+	"type": "object",
+	"properties": {
+		"enabled": { "type": "boolean" },
+		"addrs": { "type": "array", "items": { "type": "string" } },
+		"master_name": { "type": "string" }
+	}
+}
+`)
+
+type RedisSentinelConfig struct {
+	Enabled    bool     `json:"enabled,omitempty"`
+	Addrs      []string `json:"addrs,omitempty"`
+	MasterName string   `json:"master_name,omitempty"`
+}
+
+var _ = SecretConfigSchema.Add("SMTPMode", `
+{
+	"type": "string",
+	"enum": ["normal", "ssl"]
+}
+`)
+
+type SMTPMode string
+
+const (
+	SMTPModeNormal SMTPMode = "normal"
+	SMTPModeSSL    SMTPMode = "ssl"
+)
+
+var _ = SecretConfigSchema.Add("SMTPServerCredentials", `
+{
+	"type": "object",
+	"properties": {
+		"host": { "type": "string" },
+		"port": { "type": "integer", "minimum": 1, "maximum": 65535 },
+		"mode": { "$ref": "#/$defs/SMTPMode" },
+		"username": { "type": "string" },
+		"password": { "type": "string" }
+	},
+	"required": ["host", "port"]
+}
+`)
+
+type SMTPServerCredentials struct {
+	Host     string   `json:"host,omitempty"`
+	Port     int      `json:"port,omitempty"`
+	Mode     SMTPMode `json:"mode,omitempty"`
+	Username string   `json:"username,omitempty"`
+	Password string   `json:"password,omitempty"`
+}
+
+func (c *SMTPServerCredentials) SetDefaults() {
+	if c.Mode == "" {
+		c.Mode = SMTPModeNormal
```

Comment:
Default should be ssl

Reply:
It's the same default value as current configuration, do you want to change it?